mne/io/eyelink/tests/test_eyelink.py

```diff
@@ -246,6 +268,22 @@ def _simulate_eye_tracking_data(in_file, out_file):
                 "...\t1497\t5189\t512.5\t.............\n"
             )
 
+        for timestamp in np.arange(5488500, 5488600):  # 100 samples
+            fp.write(
+                f"{timestamp}\t-2434.0\t-1760.0\t840.0\t100\t20\t45\t45\t127.0\t"
+                "...\t1497\t5189\t512.5\t.............\n"
+            )
+            # Check and insert button events at this timestamp
+            while (
+                button_idx < len(button_events)
+                and button_events[button_idx][0] == timestamp
+            ):
```

Comment:
The `while` statement works, but you don't really need a _loop_ here, you can just use a conditional statement:

```suggestion
            if button_idx < len(button_events) and button_events[button_idx][0] == timestamp:
```